lib/api/api.go

```diff
@@ -448,7 +468,7 @@ func (s *service) Serve(ctx context.Context) error {
 	}
 	// Give it a moment to shut down gracefully, e.g. if we are restarting
 	// due to a config change through the API, let that finish successfully.
-	timeout, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+	timeout, cancel := context.WithTimeout(context.Background(), s.shutdownTimeout)
```

Comment:
How critical is it if this times out? Just wondering as you increased it for github actions, and amazingly slow setups also exist in reality. If it is problematic, lets at least add logging to indicate timeout happened as a clue if we ever get to debug user issue related to it in the future.

Reply:
I don't think I understand the question - I didn't change the timeout for normal operation, I just made it configurable so that the tests can override it.

Reply:
Oh I get that. I am asking about what would happen if the same condition occurred as in the test in a real setup, i.e. if this timeout is reached before webauthn does whatever it should do within the timeout. I guess it might be saving the state to db? And if it times out before, the state is simply lost (e.g. token registration)? If anything like this might happen, I am suggesting to add logging so that could be detected.

Reply:
As far as I know there isn't anything new that needs to be done during shutdown. `webauthnService` reads persistent state from disk when it's needed and writes it back down as soon as it's done with it, precisely to avoid that kind of problem. The only volatile state in `webauthnService` is authentication/registration challenges and pending successful registrations kept in "limbo" until the user chooses to save them (so that they behave like the rest of the settings GUI).

The issue I was having wasn't that some shutdown operation was interrupted, but that HTTP request handlers that changed GUI config were interrupted by the on-config-change-shutdown before they had time to write their response - the grace period inhibiting shutdown while HTTP handlers finish wasn't long enough.

I think it's just a matter of there being a lot of tests running in parallel that's causing things to run slow on GHA sometimes, and maybe `TestWebauthnConfigChanges` is our first test case that's modifying GUI config during the test.